pkg/experiment/api/experiment.go

```diff
@@ -122,15 +122,25 @@ func (s *experimentService) ListExperiments(
 	if req.FeatureVersion != nil {
 		whereParts = append(whereParts, mysql.NewFilter("feature_version", "=", req.FeatureVersion.Value))
 	}
-	if req.From != 0 {
-		whereParts = append(whereParts, mysql.NewFilter("stopped_at", ">=", req.From))
-	}
-	if req.To != 0 {
-		whereParts = append(whereParts, mysql.NewFilter("start_at", "<=", req.To))
-	}
-	if req.Status != nil {
-		whereParts = append(whereParts, mysql.NewFilter("status", "=", req.Status.Value))
-	} else if len(req.Statuses) > 0 {
+	if req.StartAt != 0 {
+		// When a start timestamp is provided,
+		// use it as the lower bound for filtering.
+		whereParts = append(whereParts, mysql.NewFilter("start_at", ">=", req.StartAt))
+	}
+	if req.StopAt != 0 {
+		// When a stop timestamp is provided:
+		// - If req.StartAt is also provided, treat req.StopAt as an absolute upper bound.
+		// (This selects experiments with stop_at <= req.StopAt.)
+		// - If req.StartAt is not provided, treat req.StopAt as a relative cutoff timestamp.
+		// (This selects experiments with stop_at >= req.StopAt.)
+		// It treats it as a relative duration when the `startAt` is not provide
+		operator := ">="
+		if req.StartAt != 0 {
+			operator = "<="
+		}
+		whereParts = append(whereParts, mysql.NewFilter("stop_at", operator, req.StopAt))
+	}
```

Comment:
Supporting this will simplify the code listing the experiment in many places.

Reference:
https://github.com/bucketeer-io/bucketeer/pull/1568/files#diff-31ab3c81724cb69fe0a25938a702c5cbdee330c4f14f515e74850c018d515d8fL344-L355